### PR TITLE
feat: added api extendibility

### PR DESCRIPTION
**Pull request by antoniomuso**
## 🚨 Proposed changes
This PR exports the API ref and all types to allow API extensibility. 

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

### CODE REVIEW

**Review by simonecorsi — APPROVED** [review]
lgtm 🚀

File: README.md [hunk 1]
```diff
@@ -122,6 +122,45 @@ spec:
 
 **Note:** `spec.type` can take values in ['website','library','service'] but to render GitLab Entity, Catalog must be of type `service`
 
+## Old/New Gitlab Versions
+
+If you have an old Gitlab version, or a new one, we allow you to extend the Gitlab Client as follow:
```
Comment (JellyBellyDev) [inline README.md:127]: GitLab

File: README.md [hunk 2]
```diff
@@ -122,6 +122,45 @@ spec:
...
+        factory: ({ configApi, discoveryApi }) =>
+            new CustomGitlabCIClient({
+                discoveryApi,
+                baseUrl: configApi.getOptionalString('gitlab.baseUrl'),
+            }),
+    }),
+];
+```
+
+`myCustomClient.ts`
```
Comment (JellyBellyDev) [inline README.md:147]: where i put this file? :)

Comment (antoniomuso) [reply README.md:147]: This can be put where you want.

Comment (antoniomuso) [reply README.md:147]: The precedenting file is a backstager file this is not.

File: README.md [hunk 3]
```diff
@@ -122,6 +122,45 @@ spec:
...
+    async getPipelineSummary(projectID: string | undefined): Promise<PipelineSummary | undefined> {
+        this.callApi(...)
+        .
+        .
+        .
+    }
+}
+```
+see [here](./src/api/GitlabCIClient.ts).
+
```
[the remark this answers is not in the document]
Comment (antoniomuso) [reply README.md:163]: ok

File: README.md (hunk 1 above)
Comment (antoniomuso) [reply README.md]: DoNE!

File: README.md [hunk 4]
```diff
@@ -122,6 +122,45 @@ spec:
 
 **Note:** `spec.type` can take values in ['website','library','service'] but to render GitLab Entity, Catalog must be of type `service`
 
+## Old/New Gitlab Versions
+
+If you have an old Gitlab version, or a new one, we allow you to extend the GitLab Client as follow:
```
Comment (JellyBellyDev) [inline README.md:127]: GitLab

File: README.md [hunk 5]
```diff
@@ -122,6 +122,45 @@ spec:
 
 **Note:** `spec.type` can take values in ['website','library','service'] but to render GitLab Entity, Catalog must be of type `service`
 
+## Old/New Gitlab Versions
```
Comment (JellyBellyDev) [inline README.md:125]: GitLab

File: README.md (hunk 2 above)
Comment (antoniomuso) [reply README.md]: I added a general path.

**Review by JellyBellyDev — APPROVED** [review]
LGTM

**Review by dnlup — APPROVED** [review]
🚀

**Pull request merged**